UI/backend/app.py

```diff
@@ -1,116 +1,131 @@
+import requests
+import json
+import pyuniversity
 from flask import Flask, request, jsonify
 from flask_cors import CORS
-import subprocess
-import os
+from constants import UniJsonKeys
+from dbUtils import connectDB, insertTimeSlot, insertCourse, insertInstructor
+
 
 app = Flask(__name__)
 CORS(app)
 
-courses = []
+def tsProvided(timeSlot, insertFuncArgs):
+    tsMembers = timeSlot.split()
+    if len(tsMembers) != 3:
+        return jsonify({'success': False, 'message': 'Invalid time slot format'}), \
+            requests.codes.bad_request
+    day, startTime, endTime = tsMembers
+    insertFuncArgs.extend([day, startTime, endTime])
+    insertTimeSlot(day, startTime, endTime)
 
 @app.route('/addCourse', methods=['POST'])
 def add_course():
     data = request.json
-    course_name = data.get('courseName', '')
-    time_slot = data.get('timeSlotCourse', '')
-    if not course_name:
-        return jsonify({'success': False, 'message': 'Course name is required'}), 400
-    
-    print(f"Received data: {data}")
-
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
+    courseName = data.get(UniJsonKeys.COURSE_NAME, '')
+    timeSlot = data.get(UniJsonKeys.TIME_SLOT_COURSE, '')
+    if not courseName:
+        return jsonify({'success': False, 'message': 'Course name is required'}), \
+            requests.codes.bad_request
+    insertFuncArgs = [courseName]
+    if timeSlot:
+        tsProvided(timeSlot, insertFuncArgs)
     try:
-        if not time_slot:
-            result = subprocess.run(['bash', cpp_exec_path, '--addCourse', course_name], capture_output=True, text=True)
-            print(f"Subprocess error: {result.stderr}")
-        else:
-            result = subprocess.run(['bash', cpp_exec_path, '--addCourse', course_name, time_slot], capture_output=True, text=True)
-
-        if result.returncode == 0:
-            return jsonify({'success': True, 'message': 'Course added successfully!'})
-        else:
-            return jsonify({'success': False, 'message': 'Failed to add course', 'details': result.stderr}), 500
-    except Exception as e:
-        print(f"Exception occurred: {str(e)}")
-        return jsonify({'success': False, 'message': str(e)}), 500
-
+        insertCourse(*insertFuncArgs)
```

Comment:
why do we need '*' before the argument?

Reply:
'*'unpack the list into individual arguments so that  it will be the actual value, not a list.